iotileship/iotile/ship/actions/ModifyJsonStep.py

```diff
@@ -0,0 +1,89 @@
+import json
+import copy
+
+from iotile.ship.exceptions import RecipeActionMissingParameter
+
+def modify_dict(data, key, value, create_if_missing=False):
+    """
+    Change (or add) a json key/value pair.
+
+    Args:
+        data (dict): The original data. This will not be modified.
+        key (str|list): Either a string or a list of keys and subkeys specifing the key to change
```

Comment:
Is there an external constraint that makes us have a multi-type argument to a function? This looks a bit complicated - shouldn't it always be a list, just sometimes with a length of one?

Reply:
Yeah, that makes the most sense. I'll change it to only accept lists.